hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-2.x-replication/src/test/java/com/google/cloud/bigtable/hbase2_x/replication/HbaseToCloudBigtableReplicationEndpointTest.java

```diff
@@ -425,4 +411,78 @@ public void testWriteFailureToBigtableDoesNotStallReplication()
     Assert.assertEquals(0, actualCells.get(1).getTimestamp());
   }
+
+  @Test
+  public void testMutationReplicationWithWALEntryFilter() throws IOException, InterruptedException {
+    Put put1 = new Put(FILTERED_ROW_KEY);
+    put1.addColumn(TestUtils.CF1, TestUtils.COL_QUALIFIER, 0, FILTERED_ROW_KEY);
+    put1.addColumn(TestUtils.CF2, TestUtils.COL_QUALIFIER, 0, FILTERED_ROW_KEY);
+    hbaseTable.put(put1);
+
+    Put put2 = new Put(ROW_KEY);
+    put2.addColumn(TestUtils.CF1, TestUtils.COL_QUALIFIER, 0, ROW_KEY);
+    put2.addColumn(TestUtils.CF2, TestUtils.COL_QUALIFIER, 0, ROW_KEY);
+    hbaseTable.put(put2);
+
+    TestUtils.waitForReplication(
+        () -> {
+          //  replicate Entries will be zero
+          return TestReplicationEndpoint.replicatedEntries.get() >= 1;
+        });
+
+    Result cbtResult = cbtTable.get(new Get(FILTERED_ROW_KEY).setMaxVersions());
+    Result hbaseResult = hbaseTable.get(new Get(FILTERED_ROW_KEY).setMaxVersions());
+    Assert.assertTrue(cbtResult.isEmpty());
+    Assert.assertFalse(hbaseResult.isEmpty());
+    Assert.assertEquals(
+        "Number of cells , actual cells: " + hbaseResult.listCells(),
+        2,
+        2,
+        hbaseResult.listCells().size());
+
+    Result cbtResult2 = cbtTable.get(new Get(ROW_KEY).setMaxVersions());
+    Result hbaseResult2 = hbaseTable.get(new Get(ROW_KEY).setMaxVersions());
+    TestUtils.assertEquals(cbtResult2, hbaseResult2);
+  }
+
+  @Test
+  public void testReplicationWithScopeFilter() throws IOException, InterruptedException {
+    TableName tableName = TableName.valueOf(UUID.randomUUID().toString());
+    createTables(
+        tableName, HConstants.REPLICATION_SCOPE_GLOBAL, HConstants.REPLICATION_SCOPE_LOCAL);
+    hbaseTable = hbaseConnection.getTable(tableName);
+    cbtTable = cbtConnection.getTable(tableName);
+
+    Put put1 = new Put(ROW_KEY);
+    put1.addColumn(TestUtils.CF1, TestUtils.COL_QUALIFIER, 0, ROW_KEY);
+    put1.addColumn(TestUtils.CF2, TestUtils.COL_QUALIFIER, 0, ROW_KEY);
+    hbaseTable.put(put1);
+
+    TestUtils.waitForReplication(
+        () -> {
+          //  replicate Entries will be 1
+          return TestReplicationEndpoint.replicatedEntries.get() >= 1;
+        });
+
+    Result cbtResult = cbtTable.get(new Get(ROW_KEY).setMaxVersions());
+    Result hbaseResult = hbaseTable.get(new Get(ROW_KEY).setMaxVersions());
+    Assert.assertFalse(cbtResult.isEmpty());
+    Assert.assertFalse(hbaseResult.isEmpty());
+    Assert.assertEquals(
+        "Number of cells , actual cells: " + hbaseResult.listCells(),
+        2,
+        2,
+        hbaseResult.listCells().size());
+
+    Assert.assertEquals(
+        "Number of cells , actual cells: " + hbaseResult.listCells(),
+        1,
+        1,
+        cbtResult.listCells().size());
+    TestUtils.assertEquals(
+        hbaseResult,
+        cbtResult,
+        HConstants.REPLICATION_SCOPE_GLOBAL,
```

Comment:
check it in test. Use assertEqual on cells.

Reply:
Done.